synapse/tools/storm.py

```diff
@@ -215,6 +218,203 @@ async def runCmdOpts(self, opts):
+
+            self._forms.append((formname, f'[form] {formname}{formdoc}'))
+
+            for prop in form['props'].values():
+                propname = prop['name']
+                if not propname.startswith('.'):
+                    propname = f':{propname}'
+
+                propdoc = prop.get('doc', '')
+                if propdoc:
+                    propdoc = f' - {propdoc}'
+
+                self._props.append((f'{formname}{propname}', f'[prop] {formname}{propname}{propdoc}'))
+
+        # Process cmds
+        commands = info['stormdocs']['commands']
+        for command in commands:
+            name = command['name']
+            doc = command['doc']
+            if doc:
+                doc = f' - {doc}'
+            self._cmds.append((name, f'[cmd] {name}{doc}'))
+
+        # Process libs
+        libraries = info['stormdocs']['libraries']
+        for library in libraries:
+            basename = '.'.join(library['path'])
+
+            for local in library['locals']:
+                libname = '.'.join((basename, local['name']))
+                name = libname
+                desc = local['desc'].strip().split('\n')[0]
+                libtype = local['type']
+                if isinstance(libtype, dict) and local['type']['type'] == 'function':
+                    args = local['type'].get('args')
+                    if args:
+                        params = []
+                        for arg in args:
+                            argname = arg['name']
+                            argtype = arg['type']
+
+                            params.append(f'{argname}: {argtype}')
+
+                        params = ', '.join(params)
+                        name = f'{name}({params})'
+                    else:
+                        name = f'{name}()'
+
+                self._libs.append((libname, f'[lib] ${name} - {desc}'))
+
+        self._forms.sort(key=lambda x: x[0])
+        self._props.sort(key=lambda x: x[0])
+        self._cmds.sort(key=lambda x: x[0])
+        self._libs.sort(key=lambda x: x[0])
+
+        self.initialized = True
+
+    async def _get_tag_completions(self, prefix='', limit=100):
+        if not prefix:
+            depth = 1
+        else:
+            depth = prefix.count('.') + 1
+
+        q = '''
+        $rslt = $lib.list()
+        if ($prefix != '') { syn:tag=$lib.regex.replace("\\.$", '', $prefix) }
+        syn:tag^=$prefix
+        +:depth<=$depth
+        | uniq
+        | limit $limit
+        | $leaf = $lib.true { -> syn:tag:up | limit 1 | $leaf = $lib.false }
+        $doc = ''
+        if $node.props.doc {
+            $doc = ` - {$node.props.doc}`
+        }
+        $rslt.append(($node.value(), `[tag] {$node.value()}{$doc}`))
+        | spin
+        | return($rslt)
+        '''
+        opts = {'vars': {'prefix': prefix, 'limit': limit, 'depth': depth}}
+        return await self._cli.item.callStorm(q, opts=opts)
+
+    def get_completions(self, document, complete_event):
```

Comment:
Not sure if it effects coverage or not, but i'd probably put the `#pragma: no cover` on the decl line rather than the pass line here.